app/code/Magento/InventoryApi/Api/GetProductQuantityInStockInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api;
+
+/**
+ * Service which returns Quantity of products available to be sold by Product SKU and Stock Id
+ *
+ * @api
+ */
+interface GetProductQuantityInStockInterface
```

Comment:
>whether naming is correct;

It's a good question
When I proposed a name for this service - GetProductQuantityInStockInterface
I meant that this service will "retrieve product quantity which is available (in stock)"
We use this naming currently (In stock/ Out of stock).

But, having "Stock" entity introduced by MSI.
The service name could be "read" differently. Like - get product quantity for specified Stock (Stock entity).

Currently, this service accepts `$stockId` as one of the parameters
`public function execute(string $sku, int $stockId): float;`
Thus, following this signature - service returns Qty for provided `$stockId`

Based on the above IMO current naming - `GetProductQuantityInStockInterface` is appropriate.
But if you have objections - let's discuss and re-name one

Reply:
I agree, here we have a term "stock" that violates the "ubiquitous language" DDD principle because in the same context it can mean both the "available quantity" (more high-level) and the "virtual aggregation" (more low-level);

probably we should think about it;

what's more, as I introduced during our last weekly meeting, in the future, when we will have the sales channel connected to our stock (virtual aggregation), maybe this service could be better named `GetProductQuantityInSalesChannelInterface` but at the moment it is good enough to leave it as is

Reply:
@aleron75  you made a very true statement regarding violation of  "ubiquitous language".
There is definitely an issue with this naming.

Regarding `GetProductQuantityInSalesChannelInterface` I thought about this. 
But came up to the conclusion that naming like `InSalesChannelInterface` sounds like a "leaky abstraction".
As I am not sure that on the level of this interface we should expose a knowledge that we have a relationship `SalesChannel` to `Stock` and that it's enough just to provide `$stockId` to get Product Qty for your Sales Channel. 
Moreover potentially we could have the single Stock being re-usable by different Sales channels.
For example, there are 3 websites:
website A, website B, website C
and we have the only stock - Stock A
Thus, all of those websites will use the same stock 

So, semantically it's incorrect to say - give me product quantity for current Sales Channel providing $stockId as the parameter.

Does it make sense?

Reply:
>semantically it's incorrect to say - give me product quantity for current Sales Channel providing $stockId as the parameter

you're perfectly right, it should be instead:

"give me product quantity for current Sales Channel providing $channelId as the parameter"

where the $channelId should be derived from the website in the current context;

by the way at the moment let's leave the service as is and just postpone this discussion when we'll work on the channel entity; thank you

Reply:
@aleron75 100% agree, let's see how our interfaces will evolve and later align naming accordingly.

Reply:
My opinion is to follow YAGNI principle.
Now we do not have the clear vision about sales channel abstraction. How to do this extensible and clear...
And we have chance to create service which will not be related to our future implementation

So we should not create service related to Sales Channels. We will do it when we will work on this task.
I believe that we could create some prototype (how to work with Sales Channels) in separate branch
But we need to keep our develop branch clear